### PR TITLE
Remove sampling from GraphTransformerNet forward

### LINKED ISSUE #102: Fix the confusing variable name:`zero_var`
`zero_var` is confusing because it doesn't zero the returned `log_var` (we had to explain this in the docstring), and it doesn't zero the variance used for anything. Instead, use `sample: bool = True`—this inverts the situation and makes it more intuitive what's going on. Set it to `False` by default.

It exists only in `model.py` and `test_model.py`, so the changes should be small and easy.

### LINKED ISSUE #104: Remove sampling from `forward`
**NOTE**: This issue invalidates #102 
---------------------

`forward` samples `pred = mu + exp(0.5*log_var)*eps` (`model.py:320`) but nothing trains `log_var`. Under a point loss the optimum is `sigma=0`, so it slides to the `-10` clamp (`model.py:317`) where gradient is 0 and the head dies. 

## Fix

```python
mu = self.mu_mlp(g)
log_var = torch.clamp(self.log_var_mlp(g), min=-10.0, max=10.0)
pred = mu
```

Drop `zero_var` (no caller uses it). Update `test_model.py:227` and the docstrings.
Keep the clamp and the `(pred, log_var)` return.

## Docs

Notebooks aren't version-compatible, so NLL training stays caller-side. 

Add to `README.md`:

````markdown
### Uncertainty estimates

`log_var` gets no gradient from a plain regression loss:

```python
loss = your_loss(pred, y) + F.gaussian_nll_loss(pred.detach(), y, log_var.exp())
```

`.detach()` keeps the NLL off the mean: NLL trains only `log_var`, and `mu` stays on the existing point-loss mix.
 Masked: `reduction='none'`, then `(nll * valid).sum() / valid.sum().clamp_min(1)`. 
For data with missing values, as in notebooks, `valid = mask.bool() & torch.isfinite(y) & torch.isfinite(pred)`.
At eval `sigma = (0.5*log_var).exp()` is a real error bar.
```
````

### PR DESCRIPTION
**Pull request by mckinley-rezo**
Removes sampling from the GraphTransformerNet forward. 

This change fixes the issue where the predictions were sampled using `prediction = mu + std * epsilon`. This means, for point losses (like MSE), where the sampling is increasing the loss value by the variance, it will minimize loss when std  == 0. This only happens when `log_var` approaches -inf, causing the value to go beyond the -10 clamp, and the gradient for the `log_var` head to go to 0.

This PR removes sampling entirely, meaning that this function always returns the mean as the prediction (`pred = mu`), while also returning `log_var` for variance aware loss or uncertainty.

closes #104
closes #102

### CODE REVIEW

File: gt_pyg/nn/model.py [hunk 1]
```diff
@@ -313,15 +298,8 @@ def forward(
...
-        # Numerical stability: clamp log_var range a bit
         log_var = torch.clamp(log_var, min=-10.0, max=10.0)
-        std = torch.exp(0.5 * log_var)  # [B, T]
-
-        if self.training and not zero_var:
-            eps = torch.randn_like(std)
-            pred = mu + std * eps       # reparameterized sample
-        else:
-            pred = mu                   # deterministic mean
+        pred = mu
```
Comment (pawel-rezo) [inline gt_pyg/nn/model.py]: This assignment is redundant

**Review by pawel-rezo — COMMENTED** [review]
This looks good.

File: gt_pyg/nn/model.py (hunk 1 above)
Comment (mckinley-rezo) [reply gt_pyg/nn/model.py]: SG, made both declarations into a single line

**Review by pawel-rezo — APPROVED** [review]
